internal/controller/tenant_controller.go

```diff
@@ -557,12 +551,40 @@ func (r *TenantReconciler) reconcileArgoCD(ctx context.Context, tenant *cattagev
 	proj.SetLabels(map[string]string{
 		constants.OwnerTenant: tenant.Name,
 	})
+	val, found, err := unstructured.NestedSlice(proj.UnstructuredContent(), "spec", "syncWindows")
+	if err != nil {
+		return err
+	}
+	var syncWindows cattagev1beta1.SyncWindows
+	if !found {
+		syncWindows = cattagev1beta1.SyncWindows{}
+	} else {
+		syncWindows, err = fromUnstructuredSlice[cattagev1beta1.SyncWindows](val)
+		if err != nil {
+			return err
+		}
+	}
+	swResources, sws, err := r.getSyncWindows(ctx, tenant.Name)
+	if err != nil {
+		return fmt.Errorf("failed to get sync windows: %w", err)
+	}
+	syncWindows = append(syncWindows, sws...)
+	if len(syncWindows) != 0 {
+		ret, err := toUnstructuredSlice[cattagev1beta1.SyncWindows](syncWindows)
```

Comment:
```suggestion
		ret, err := toUnstructuredSlice(syncWindows)
```

[NITS]
The type can be inferred by the argument `syncWindows`, so it can be omitted.